sdk/core/amqp-common/src/errors.ts

```diff
@@ -597,8 +597,8 @@ export function translate(err: AmqpError | Error): MessagingError {
     error = new MessagingError(description);
     if ((err as any).stack) error.stack = (err as any).stack;
     if (condition) {
-      const amqpErrorCondition = SystemErrorConditionMapper[condition as any];
-      error.name = ConditionErrorNameMapper[amqpErrorCondition as any];
+      const amqpErrorCondition = (<any>SystemErrorConditionMapper)[condition];
+      error.name = (<any>ConditionErrorNameMapper)[amqpErrorCondition];
```

Comment:
This was solved a little differently for core-amqp in https://github.com/Azure/azure-sdk-for-js/pull/4928/files

@chradek Any preference for one over the other?

Reply:
Definitely prefer anything that's more specific than any!